addons/markers/XEH_postInit.sqf

```diff
@@ -1,20 +1,77 @@
-// by commy2
 #include "script_component.hpp"
 
-// recieve remote marker data
-[QGVAR(setMarkerNetwork), {_this call DFUNC(setMarkerNetwork)}] call EFUNC(common,addEventHandler);
+[QGVAR(loadMapData), DFUNC(loadMapData)] call EFUNC(common,addEventHandler);
+[QGVAR(setMarkerGlobal), DFUNC(setMarkerGlobal)] call EFUNC(common,addEventHandler);
+[QGVAR(loadBriefingMarkers), DFUNC(loadBriefingMarkers)] call EFUNC(common,addEventHandler);
+[QGVAR(onMapItemChange), DFUNC(onMapChange)] call EFUNC(common,addEventHandler);
 
-// recieve marker data for JIP
-[QGVAR(setMarkerJIP), {_this call DFUNC(setMarkerJIP)}] call EFUNC(common,addEventHandler);
+if(hasInterface) then {
+    //request map data from server
+    ["playerChanged", {
+        GVAR(currentMapID) = -1;
+        private _assignedMap = ACE_player call FUNC(getAssignedMap);
+        GVAR(currentMap) = _assignedMap;
+        [QGVAR(onMapItemChange), [_assignedMap]] call EFUNC(common,localEvent);
+    }] call EFUNC(common,addEventHandler);
 
-// request marker data for JIP
-if (isMultiplayer && {!isServer} && {hasInterface}) then {
-    GVAR(localLogic) = (createGroup sideLogic) createUnit ["Logic", [0,0,0], [], 0, "NONE"];
-    [QGVAR(sendMarkersJIP), [GVAR(localLogic)]] call EFUNC(common,serverEvent);
+    //request map data from server on each inventory change (after first request is done for this ACE_player)
+    ["playerInventoryChanged", {
+        if (!GVAR(enableUniqueMaps) || {GVAR(currentMapID) < 0}) exitWith {};
+        private _assignedMap = ACE_player call FUNC(getAssignedMap);
+        if (GVAR(currentMap) != _assignedMap) then {
+            GVAR(currentMap) = _assignedMap;
+            [QGVAR(onMapItemChange), [_assignedMap]] call EFUNC(common,localEvent);
+        };
+    }] call EFUNC(common,addEventHandler);
+
+    //check marker changes after map is closed
+    ["visibleMapChanged", {
+        params ["", "_mapOn"];
+        TRACE_1("visibleMapChanged",_mapOn);
+
+        if (!GVAR(enableUniqueMaps)) exitWith {};
+        if (_mapOn) then {
+            GVAR(newMapMarkers) = [];
+        } else {
+            private _assignedMap = ACE_player call FUNC(getAssignedMap);
```

Comment:
This code branch should also run in case somehow the current map item changes while you have the map opened (probably from the `playerInventoryChanged` event above). Hence I recommend moving the code to a dedicated function and running it from both places.